trikot-streams/streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/processors/CollectProcessor.kt

```diff
@@ -0,0 +1,39 @@
+import com.mirego.trikot.foundation.concurrent.atomic
+import com.mirego.trikot.streams.reactive.StreamsProcessorException
+import com.mirego.trikot.streams.reactive.processors.AbstractProcessor
+import com.mirego.trikot.streams.reactive.processors.ProcessorSubscription
+import org.reactivestreams.Publisher
+import org.reactivestreams.Subscriber
+
+typealias CollectorBlock<T, R> = (R, T) -> R
+
+class CollectProcessor<T, R>(
+    parentPublisher: Publisher<T>,
+    private val seed: R,
```

Comment:
I think we need to provide a supplier, not the value itself.
`private val supplier: () -> R`

The issue here is that the value is kept at the processor level, while it should be created during subscription and passed to the subscription. Each new subscription should have a new seed created.